File: frontend/src/api/consent.ts
import { api } from './client'

export interface ConsentText {
  version: string
  text_hash: string
  body_markdown: string
}

export interface Consent {
  id: string
  user_id: string
  version: string
  text_hash: string
  signature_hash: string
  created_at: string
}

export const consentApi = {
  current: () => api.get<ConsentText>('/consent/current'),
  accept: (version: string, text_hash: string) =>
    api.post<Consent>('/consent/accept', { version, text_hash }),
}

File: backend/app/schemas/consent.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ConsentTextOut(BaseModel):
    version: str
    text_hash: str
    body_markdown: str
    body_html: str


class ConsentAcceptIn(BaseModel):
    version: str
    text_hash: str  # client must echo what they actually saw


class ConsentOut(BaseModel):
    id: UUID
    user_id: UUID
    version: str
    text_hash: str
    signature_hash: str
    created_at: datetime
